orc8r/cloud/go/services/ctraced/obsidian/handlers/handlers.go

```diff
@@ -0,0 +1,246 @@
+/*
+ * Copyright 2020 The Magma Authors.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package handlers
+
+import (
+	"net/http"
+
+	"magma/orc8r/cloud/go/obsidian"
+	"magma/orc8r/cloud/go/orc8r"
+	"magma/orc8r/cloud/go/serdes"
+	"magma/orc8r/cloud/go/services/configurator"
+	"magma/orc8r/cloud/go/services/ctraced/obsidian/models"
+	"magma/orc8r/cloud/go/services/ctraced/storage"
+	merrors "magma/orc8r/lib/go/errors"
+	"magma/orc8r/lib/go/protos"
+
+	"github.com/labstack/echo"
+	"github.com/pkg/errors"
+)
+
+const (
+	Tracing = "tracing"
+	// v1/networks/:network_id/tracing
+	TracingRootPath = obsidian.V1Root + obsidian.MagmaNetworksUrlPart + obsidian.UrlSep + ":" + pathParamNetworkID + obsidian.UrlSep + Tracing
+	// v1/networks/:network_id/tracing/:trace_id
+	TracingPath = TracingRootPath + obsidian.UrlSep + ":" + pathParamTraceID
+	// v1/networks/:network_id/tracing/:trace_id/download
+	TracingDownloadPath = TracingPath + obsidian.UrlSep + "download"
+
+	pathParamTraceID   = "trace_id"
+	pathParamNetworkID = "network_id"
+)
+
+func GetObsidianHandlers(client GWCtracedClient, storage storage.CtracedStorage) []obsidian.Handler {
+	//client := NewGwCtracedClientImpl()
+	ret := []obsidian.Handler{
+		{Path: TracingRootPath, Methods: obsidian.GET, HandlerFunc: ListCallTraces},
+		{Path: TracingRootPath, Methods: obsidian.POST, HandlerFunc: GetCreateCallTraceHandlerFunc(client)},
+		{Path: TracingPath, Methods: obsidian.GET, HandlerFunc: GetCallTrace},
+		{Path: TracingPath, Methods: obsidian.PUT, HandlerFunc: GetUpdateCallTraceHandlerFunc(client, storage)},
+		{Path: TracingPath, Methods: obsidian.DELETE, HandlerFunc: GetDeleteCallTraceHandlerFunc(client, storage)},
+		{Path: TracingDownloadPath, Methods: obsidian.GET, HandlerFunc: GetDownloadCallTraceHandlerFunc(storage)},
+	}
+
+	return ret
+}
+
+func ListCallTraces(c echo.Context) error {
```

Comment:
make all the methods local to this file as private.

Reply:
Okay